app/upgrade-poolChange.go

```diff
@@ -0,0 +1,96 @@
+
+		allPools := app.clpKeeper.GetPools(ctx)
+		lps := clp.LiquidityProviders{}
+		poolList := clp.Pools{}
+		hasError := false
+		for _, pool := range allPools {
+			lpList := app.clpKeeper.GetLiquidityProvidersForAsset(ctx, pool.ExternalAsset)
+			temp := sdk.ZeroUint()
+			tempExternal := sdk.ZeroUint()
+			tempNative := sdk.ZeroUint()
+			for _, lp := range lpList {
+				withdrawNativeAssetAmount, withdrawExternalAssetAmount, _, _ := clp.CalculateWithdrawal(pool.PoolUnits, pool.NativeAssetBalance.String(),
+					pool.ExternalAssetBalance.String(), lp.LiquidityProviderUnits.String(), sdk.NewUint(clp.MaxWbasis).String(), sdk.NewInt(0))
+				newLpUnits, lpUnits, err := clp.CalculatePoolUnits(pool.ExternalAsset.Symbol, temp, tempNative, tempExternal,
+					withdrawNativeAssetAmount, withdrawExternalAssetAmount)
+				if err != nil {
+					hasError = true
+					ctx.Logger().Error(fmt.Sprintf("failed to calculate pool units for | Pool : %s | LP %s ", pool.String(), lp.String()))
+					break
+				}
+				lp.LiquidityProviderUnits = lpUnits
+				if !lp.Validate() {
+					hasError = true
+					ctx.Logger().Error(fmt.Sprintf("Invalid | LP %s ", lp.String()))
+					break
+				}
+				lps = append(lps, lp)
+				tempExternal = tempExternal.Add(withdrawExternalAssetAmount)
+				tempNative = tempNative.Add(withdrawNativeAssetAmount)
+				temp = newLpUnits
+			}
+			pool.PoolUnits = temp
+			if !app.clpKeeper.ValidatePool(pool) {
+				hasError = true
+				ctx.Logger().Error(fmt.Sprintf("Invalid | Pool %s ", pool.String()))
+				break
+			}
+			poolList = append(poolList, pool)
+		}
+		// If we have error dont set state
+		if hasError {
+			ctx.Logger().Error("Failed to execute upgrade plan for pool re-balance")
+		}
+		// If we have no errors , Set state .
+		if !hasError {
+			for _, pool := range poolList {
+				_ = app.clpKeeper.SetPool(ctx, pool)
+			}
+			for _, l := range lps {
+				app.clpKeeper.SetLiquidityProvider(ctx, l)
+			}
+		}
+	}
+}
+
+func ExportAppState(name string, app *SifchainApp, ctx sdk.Context) {
+	appState, vallist, err := app.ExportAppStateAndValidators(true, []string{})
+	if err != nil {
+		ctx.Logger().Error(fmt.Sprintf("failed to export app state: %s", err))
+		return
+	}
+	appStateJSON, err := app.cdc.MarshalJSON(appState)
+	if err != nil {
+		ctx.Logger().Error(fmt.Sprintf("failed to marshal application genesis state: %s", err.Error()))
+		return
+	}
+	valList, err := json.MarshalIndent(vallist, "", " ")
+	if err != nil {
+		ctx.Logger().Error(fmt.Sprintf("failed to marshal application genesis state: %s", err.Error()))
+	}
+
+	err = ioutil.WriteFile(fmt.Sprintf("%v-state.json", name), appStateJSON, 0600)
+	if err != nil {
+		ctx.Logger().Error(fmt.Sprintf("failed to write state to file: %s", err.Error()))
+	}
+	err = ioutil.WriteFile(fmt.Sprintf("%v-validator.json", name), valList, 0600)
+	if err != nil {
+		ctx.Logger().Error(fmt.Sprintf("failed to write Validator List to file: %s", err.Error()))
+	}
```

Comment:
Why do you not want to preemptively return an any non-nil error case?

Reply:
I think the idea was, to write to a whatever we can , becuase I see ExportAppState does not return an error . 
So its like write to a file if you can , if you cant just continue . 
@iverc , is that correct ?